File: algorithms/implementation/C#/CutTheSticks.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution {

    int positiveMin(int[] array)
    {
        if (array == null || array.Length == 0)
        {
            return Int32.MinValue;
        }
        var min = Int32.MaxValue;
        foreach(var element in array)
        {
            if (min > element && element > 0)
            {
                min = element;
            }
        }
        return min;
    }
    
    static void Main(String[] args) {
        int n = Convert.ToInt32(Console.ReadLine());
        string[] arr_temp = Console.ReadLine().Split(' ');
        int[] arr = Array.ConvertAll(arr_temp,Int32.Parse);
        
        var count = arr.Length;
        while (count != 0)
        {
            var solution = new Solution();
            
            var min = solution.positiveMin(arr);
            var shortened = 0;
                       
            for(var index = 0; index < arr.Length; index++)
            {
                if (arr[index] > 0)
                {
                    if (arr[index] - min <= 0)
                    {
                        count -= 1;
                    }
                    shortened += 1;
                }
                arr[index] -= min;
            }
            Console.WriteLine(shortened);          
        }
    }
}

File: algorithms/implementation/python3/cut_the_sticks.py
#!/bin/python3

import sys


n = int(input().strip())
arr = [int(arr_temp) for arr_temp in input().strip().split(' ')]

def positive_min(arr):
    if arr == None or len(arr) == 0:
        return None
    min = sys.maxsize
    for element in arr:
        if min > element and element > 0:
            min = element
    return min

count = len(arr)
while count != 0:
    shortened = 0
    min = positive_min(arr)
    for index in range(len(arr)):
        if arr[index] > 0:
            if arr[index] - min <= 0:
                count -= 1
            shortened += 1
        arr[index] -= min
    print(shortened)
